app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -99,24 +91,8 @@ def self.pulp_primary!
           pulp_primary || fail(_("Could not find a smart proxy with pulp feature."))
         end
 
-        def self.default_capsule
-          pulp_primary
-        end
-
-        def self.default_capsule!
-          pulp_primary!
-        end
-
         def self.with_environment(environment, include_default = false)
-          (pulp2_proxies_with_environment(environment, include_default) + pulpcore_proxies_with_environment(environment)).try(:uniq)
-        end
-
-        def self.pulp2_proxies_with_environment(environment, include_default = false)
-          features = [PULP_NODE_FEATURE]
-          features << PULP_FEATURE if include_default
-
-          unscoped.with_features(features).joins(:capsule_lifecycle_environments).
-            where(katello_capsule_lifecycle_environments: { lifecycle_environment_id: environment.id })
+          pulpcore_proxies_with_environment(environment).try(:uniq)
         end
 
         def self.pulpcore_proxies_with_environment(environment)
```

Comment:
I thought about optimizing this to:
```ruby
unscoped.select { |p| p.pulp_mirror? }.joins(:capsule_lifecycle_environments).
  where(katello_capsule_lifecycle_environments: { lifecycle_environment_id: environment.id }).distinct
```

Reply:
This would be fine, I like avoiding excess queries where possible.